### ISSUE TITLE
A11y: Self-collapsing mode -  Component keyboard interaction fails in some options

### ISSUE THREAD

**Issue by shopwareBot** [shopware/platform#5048]
Component keyboard interaction fails when selecting custom product options by pressing {{Enter}} key. [https://www.ibm.com/able/toolkit/verify/manual/|https://www.ibm.com/able/toolkit/verify/manual/|smart-link] 

*Steps to reproduce:* (Tested with Chrome without any screen reader)

* Create a custom product with all possible options with the self-collapsing mode activated
* Navigate to the PDP of the created custom product
* Use keyboard navigation with the {{Tab}} keyboard
* Select options with the {{Enter}} keyboard

||*Element*||*Action*||*Expected Interaction*||*Actual Interaction*||*Notes*||
|Checkbox toggle|* Navigate to toggle element 
* Press {{Enter}}|Toggle on/off|Opens file upload modal, if a file  selected uploads it to file upload element.|Pressing {{Space}} turn on/off the toggle|
|Color selection|* Navigate to an option
* Press {{Enter}}|Save selected option|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Date field|* Navigate to date selection
* Press {{Enter}}|Select current date|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Image selection|* Navigate to an option
* Press {{Enter}}|Save selected option|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Number field|* Navigate to number field
* Type a number
* Press {{Enter}}|Save typed number|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Selection field|* Navigate to an option
* Press {{Enter}}|Save selected option|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Text field|* Navigate to text field
* Type something
* Press {{Enter}}|Save typed text|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Time field|* Navigate to time field
* Type a time
* Press {{Enter}}|Save typed time|Time cannot be saved| |
|One time link toggle|* Navigate to share your config
* Press {{Enter}}|Toggle on/off|Opens file upload modal, if a file  selected uploads it to file upload element.| |
|Double check config|* Navigate to checkbox
* Press {{Enter}}|Check checkbox|Opens file upload modal, if a file  selected uploads it to file upload element.| |

h5. Make sure that...

* All actions in a widget can be performed with the keyboard
* Items can be traversed in forward and reverse order
* Standard keyboard interaction for a component is supported
* Visual indicators of focus and, where relevant, state exist
* Custom shortcuts are not accidentally triggered

**Comment by shopwareBot** [comment]
[public] Shopware automated response: This issue is linked to the internal issue: **CUS-736**



[created from CUS-736, comment 501974]
